--- yum-3.4.3/yum/config.py~	2014-06-14 16:42:42.000000000 +0300
+++ yum-3.4.3/yum/config.py	2014-06-14 16:47:26.499590981 +0300
@@ -50,10 +50,11 @@
 __main_failovermethod_default__ = 'roundrobin'
 __main_installonly_limit_default__ = 0
 __group_command_default__ = 'compat'
-__exactarchlist_default__ = ['kernel', 'kernel-smp',
-                             'kernel-hugemem', 'kernel-enterprise',
-                             'kernel-bigmem',
-                             'kernel-devel', 'kernel-PAE', 'kernel-PAE-debug']
+__kernelpkgnames_default__ = ['kernel','kernel-smp', 'kernel-longterm', 'kernel-3.4', 'kernel-3.10']
+__installonlypkgs_default__ = __kernelpkgnames_default__ + [
+    'installonlypkg(kernel)', 'installonlypkg(kernel-module)', 'installonlypkg(vm)'
+]
+__exactarchlist_default__ = []
 
 class Option(object):
     """
@@ -766,8 +767,7 @@
     installonly_limit = PositiveIntOption(__main_installonly_limit_default__,
                                           range_min=2,
                                           names_of_0=["0", "<off>"])
-    kernelpkgnames = ListOption(['kernel','kernel-smp', 'kernel-enterprise',
-            'kernel-bigmem', 'kernel-BOOT', 'kernel-PAE', 'kernel-PAE-debug'])
+    kernelpkgnames = ListOption(__kernelpkgnames_default__)
     exactarchlist = ListOption(__exactarchlist_default__)
     tsflags = ListOption()
     override_install_langs = Option()
