--- yum-3.4.3/test/check-po-yes-no.py~	2013-04-15 09:02:49.000000000 +0300
+++ yum-3.4.3/test/check-po-yes-no.py	2013-04-15 09:03:25.975416605 +0300
@@ -6,6 +6,9 @@
 import sys
 import glob
 
+import unittest
+from testbase import *
+
 # Don't import from yum, as it isn't there when we are distro. building...
 def to_utf8(obj, errors='replace'):
     '''convert 'unicode' to an encoded utf-8 byte string '''
