--- yum-3.4.3/rpmUtils/arch.py~	2013-04-15 09:00:20.000000000 +0300
+++ yum-3.4.3/rpmUtils/arch.py	2013-04-15 09:01:54.771200925 +0300
@@ -258,27 +258,9 @@
         offset = offset + fmtlen
 
 def getCanonX86Arch(arch):
-    # 
     if arch == "i586":
-        for line in _try_read_cpuinfo():
-            if line.startswith("model name"):
-                if line.find("Geode(TM)") != -1:
-                    return "geode"
-                break
-        return arch
-    # only athlon vs i686 isn't handled with uname currently
-    if arch != "i686":
-        return arch
-
-    # if we're i686 and AuthenticAMD, then we should be an athlon
-    for line in _try_read_cpuinfo():
-        if line.startswith("vendor") and line.find("AuthenticAMD") != -1:
-            return "athlon"
-        # i686 doesn't guarantee cmov, but we depend on it
-        elif line.startswith("flags"):
-            if line.find("cmov") == -1:
-                return "i586"
-            break
+        # nearest matching arch in Th is i486
+        return "i486"
 
     return arch
 
@@ -275,24 +261,6 @@
         return "sparcv8"
     return arch
 
-def getCanonX86_64Arch(arch):
-    if arch != "x86_64":
-        return arch
-
-    vendor = None
-    for line in _try_read_cpuinfo():
-        if line.startswith("vendor_id"):
-            vendor = line.split(':')[1]
-            break
-    if vendor is None:
-        return arch
-
-    if vendor.find("Authentic AMD") != -1 or vendor.find("AuthenticAMD") != -1:
-        return "amd64"
-    if vendor.find("GenuineIntel") != -1:
-        return "ia32e"
-    return arch
-        
 def getCanonArch(skipRpmPlatform = 0):
     if not skipRpmPlatform and os.access("/etc/rpm/platform", os.R_OK):
         try:
@@ -313,8 +281,6 @@
         return getCanonPPCArch(arch)
     if arch.startswith("sparc"):
         return getCanonSPARCArch(arch)
-    if arch == "x86_64":
-        return getCanonX86_64Arch(arch)
 
     return arch
